src/Microsoft.TestPlatform.ObjectModel/Client/Events/DiscoveredSource.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+
+#nullable disable
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+
+/// <summary>
+/// Describes a source file that was discovered, or that was requested to be discovered.
+/// </summary>
+public class DiscoveredSource
+{
+    [DataMember]
+    public string Source { get; set; }
+
+    public static List<DiscoveredSource> ToDiscoveredSources(List<string> sources)
```

Comment:
Also, you could take `IEnumerable<string>` instead of `List` as you are only iterating.

Reply:
Moved to DiscoveredSourceHelper internal class instead, the helper is useful in only handful of places, and the extended type is very generic.